### PR TITLE
#6Task. yulia.miasnikova

### PR DESCRIPTION
**Pull request by YuliaSirius**
https://travis-ci.org/github/YuliaSirius/js-assignments/builds/709092988

### CODE REVIEW

**Comment by AlexMokin** [comment]
Pr must include changes only in files related to the given task
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by YuliaSirius** [comment]
Please tell me if I removed commits wrong. I removed unnecessary commits with the command git reset , and add necessary commit.

**Comment by YuliaSirius** [comment]
Travis CI link:
https://travis-ci.org/github/YuliaSirius/js-assignments/builds/710310037

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -81,79 +86,85 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,1,1   =>  false
  *   10,10,10 =>  true
  */
-function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+function isTriangle(a, b, c) {
+  if (a + b > c && a + c > b && b + c > a) return true;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: just return boolean expression

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -81,79 +86,85 @@ function getSumBetweenNumbers(n1, n2) {
...
  *     width: 20,
  *     height: 10
  *  }
- * 
+ *
  *  (5;5)
- *     -------------  
- *     |           | 
+ *     -------------
+ *     |           |
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:105]: do not change formatting of the original code or comments

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -81,79 +86,85 @@ function getSumBetweenNumbers(n1, n2) {
...
- * 
+ *
  *   { top: 0, left: 0, width: 10, height: 10 },
  *   { top:20, left:20, width: 20, height: 20 }    =>  false
- *  
+ *
  */
 function doRectanglesOverlap(rect1, rect2) {
-    throw new Error('Not implemented');
+  if (
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: just return boolean expression

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -251,10 +275,24 @@ function reverseInteger(num) {
...
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+  let num, digits, flag, sum;
+  flag = true;
+  sum = 0;
+  digits = (ccn + '').split('').reverse();
+  for (let i = 0; i < digits.length; i++) {
+    num = digits[i];
+    num = parseInt(num, 10);
+    if ((flag = !flag)) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not do assignmnts inside if operator

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -293,13 +333,32 @@ function getDigitalRoot(num) {
...
-    throw new Error('Not implemented');
+  let stack = [];
+  let map = {
+    '(': ')',
+    '[': ']',
+    '{': '}',
+    '<': '>',
+  };
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === '(' || str[i] === '{' || str[i] === '[' || str[i] === '<') {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: just check map key https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Do not duplicate the list of brackets

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -332,10 +391,26 @@ function isBracketsBalanced(str) {
...
+  if (period / 60 <= 90) return 'an hour ago';
+  if (period / 60 / 60 <= 22) return `${round(period / 60 / 60)} hours ago`;
+  if (period / 60 / 60 <= 36) return 'a day ago';
+  if (period / 60 / 60 / 24 <= 25)
+    return `${round(period / 60 / 60 / 24)} days ago`;
+  if (period / 60 / 60 / 24 <= 45) return 'a month ago';
+  if (period / 60 / 60 / 24 <= 345)
+    return `${round(period / 60 / 60 / 24 / 30)} months ago`;
+  if (period / 60 / 60 / 24 <= 545) return 'a year ago';
+  if (period / 60 / 60 / 24 > 545)
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: always use curly braces if  it's not a one-line if

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -431,27 +514,45 @@ function getMatrixProduct(m1, m2) {
...
-    doRectanglesOverlap: doRectanglesOverlap,
-    isInsideCircle: isInsideCircle,
-    findFirstSingleChar: findFirstSingleChar,
-    getIntervalString : getIntervalString,
-    reverseString: reverseString,
-    reverseInteger: reverseInteger,
-    isCreditCardNumber: isCreditCardNumber,
-    getDigitalRoot: getDigitalRoot,
-    isBracketsBalanced: isBracketsBalanced,
-    timespanToHumanString : timespanToHumanString,
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:452]: do not modify formatting of the original code

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by YuliaSirius** [comment]
https://travis-ci.org/github/YuliaSirius/js-assignments/builds/711007893

**Comment by AlexMokin** [comment]
Looks good. But rhere must not be any chagnes to the original code and comments  formatting. See images below

image
image

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by YuliaSirius** [comment]
I created a new PR instead of this- #778.

**Pull request closed without merging**